src/vpn-api/utils/clients.ts

```diff
@@ -78,59 +79,50 @@ const setDeviceState = (() => {
 					}
 					// Update the current state on success
 					deviceStates[uuid].currentState = targetState;
-					// Log updated state
-					let stateMsg = `common_name=${targetState.common_name} connected=${
-						targetState.connected
-					}`;
-					if (targetState.virtual_address != null) {
-						stateMsg = `${stateMsg} virtual_address=${
-							targetState.virtual_address
-						}`;
-					}
-					logger.info(`Successfully updated state for ${uuid}: ${stateMsg}`);
+					return targetState as DeviceState;
```

Comment:
The `Partial` is required for when tracking the state of a previously unknown devices, we initialise the states as `{}`.

Reply:
You mean the
```ts
		if (deviceStates[uuid] == null) {
			deviceStates[uuid] = {
				targetState: {},
				currentState: {},
				promise: Bluebird.resolve(),
			};
		}
		deviceStates[uuid].targetState = state;
```
bit below?  In that case it can be:
```ts
		if (deviceStates[uuid] == null) {
			deviceStates[uuid] = {
				targetState: state,
				currentState: {},
				promise: Bluebird.resolve(),
			};
		} else {
			deviceStates[uuid].targetState = state;
		}
```
in order to avoid the `Partial` on `targetState`